core/templates/dev/head/i18n.js

```diff
@@ -90,6 +90,11 @@ oppia.controller('I18nFooter', [
         $http.put(siteLanguageUrl, {
           site_language_code: $scope.currentLanguageCode
         });
+      } else {
+        // This is a workaround for the library page as the base url
```

Comment:
I think this approach is reasonable, actually. But I'm a little confused about when it's applied. From the conditional, it seems to be applied when the user is not logged in (as opposed to when the user is on the library page)?